contributors/design-proposals/storage/csi-migration.md

```diff
@@ -841,6 +841,43 @@ When the above conditions are not met, the PV Controller will use FindProvisiona
 to determine the in-tree plugin that can be used for provisioning a volume (as is
 the case today).
 
+Update: 1/13/2020
+
+In Beta we discovered issue: https://github.com/kubernetes/kubernetes/issues/79043
+
+In order to resolve this the design needs to be modified. When migration is "on"
+the Persistent Volume Controller will still set the Storage Provisioner
+annotation (volume.beta.kubernetes.io/storage-provisioner) with the name of the
+migrated CSI driver. However, it will also set an additional annotation
+`volume.beta.kubernetes.io/migrated-to` to the CSI Driver name. Additionally,
+the PV Controller will be modified so that it lists all PV's and PVC's on
+startup and checks the `volume.beta.kubernetes.io/storage-provisioner` and
+`pv.kubernetes.io/provisioned-by` annotations respectively to anneal the
+required `volume.beta.kubernetes.io/migrated-to` annotation. An example is a PV
+with `volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/gce-pd`, when
+the PV controller starts up with `CSIMigrationGCE=true` the controller will
+additionally annotate the PV with
+`volume.beta.kubernetes.io/migrated-to=pd.csi.storage.gke.io`. The PV controller
+will also remove `migrated-to` annotations on PV/PVCs with migration OFF to
+support rollback scenarios.
+
+This `migrated-to` annotation will be used by `v1.6.0+` of the CSI External
+provisioner to determine whether a PV or PVC should be operated on by the
+provisioner. The benefit being that PV Controller start-up annealing of this
+annotation will allow the PV Controller to stand down and the CSI External
+Provisioner to pick up a PV that was dynamically provisioned "before" migration
+was enabled. These "newer" external provisioners will still be compatible with
+older versions of Kubernetes doing migration as they will still pick up volumes
+annotated in the original way. This change makes the implementation consistent
```

Comment:
When you say the updated external provisioner will work on existing k8s versions, do you mean that they will delete migrated volumes annotated with the in-tree driver name? Or do you mean that external provisioner will behave like older versions of external provisioner if k8s does not set the `migrated-to` annotation?

Reply:
The second statement. I will clarify in the text.